doc/creating_a_new_project.md

```diff
@@ -0,0 +1,50 @@
+# Creating a New Project
+
+## Creating a new WinRT Component DLL and referencing it in another project
+
+When creating a new DLL, it was really helpful to reference an existing DLL's `.vcxproj` like `TerminalControl.vcxproj`. While you should mostly try to copy what the existing `.vcxproj` has, here's a handful of things to double check for as you go along.
+
+- [ ] Make sure to `<Import>` our pre and post build props in your `NewDLL.vcxproj`
+```
+<!-- pre props --> 
+<Import Project="..\..\..\common.openconsole.props" Condition="'$(OpenConsoleDir)'==''" />
+<Import Project="$(OpenConsoleDir)src\cppwinrt.build.pre.props" />
+<!-- post props -->
+<Import Project="$(OpenConsoleDir)src\cppwinrt.build.post.props" />
+```
+- [ ] Add a `<ProjectReference>` to your new `.vcxproj` in both `WindowsTerminal.vcxproj` and `TerminalApp.vcxproj`
+- [ ] Add a `<Reference>` to `TerminalAppLib.vcxproj` similar to this:
+```
+    <Reference Include="Microsoft.Terminal.NewDLL">
+      <HintPath>$(_BinRoot)TerminalNewDLL\Microsoft.Terminal.NewDLL.winmd</HintPath>
+      <IsWinMDFile>true</IsWinMDFile>
+      <Private>false</Private>
+      <CopyLocalSatelliteAssemblies>false</CopyLocalSatelliteAssemblies>
+    </Reference>
+```
+- [ ] Make sure the project has a `.def` file with the following lines. The `WINRT_GetActivationFactory` part is important to expose the new DLL's activation factory so that other projects can successfully call the DLL's `GetActivationFactory` to get the DLL's classes.
+```
+EXPORTS
+DllCanUnloadNow = WINRT_CanUnloadNow                    PRIVATE
+DllGetActivationFactory = WINRT_GetActivationFactory    PRIVATE
+```
+- For a bit more context on this whole process, the `AppXManifest.xml` file defines which classes belong to which DLLs. So if your project wants class `X.Y.Z`, it can look it up in the manifest and see that it came from `X.Y.dll`. Then it'll load up the dll, and call a particular function called `GetActivationFactory(L"X.Y.Z")` to get the class it wants. 

```

Comment:
Maybe mention that the definitions in the Appxmanifest.xml are absolutely required to make this activation work. And the reason that you don't have to define them manually is because you include this library as a reference in TerminalApp which is in WindowsTerminal.exe which is already inside a Centennial Packaging project (CascadiaPackage) which will automatically enumerate the reference tree of WinMDs and stitch that information into the appxmanifest.xml. But if you're for some reason making another EXE or joining this up as a reference to something that isn't already rolling up into a packaging project who grovels for the metadata to put in appxmanifest.xml... that might become your problem.